sheep/src/format/mod.rs

```diff
@@ -1,10 +1,14 @@
+use serde::Serialize;
+
 #[cfg(feature = "amethyst")]
 pub mod amethyst;
 
+pub mod named;
+
 use SpriteAnchor;
 
 pub trait Format {
-    type Data;
+    type Data: Serialize;
 
-    fn encode(dimensions: (u32, u32), sprites: &[SpriteAnchor]) -> Self::Data;
+    fn encode(&self, dimensions: (u32, u32), sprites: &[SpriteAnchor]) -> Self::Data;
```

Comment:
Now that I think about it, with this change, it would be possible to simply pass an array of names as options for the normal `Amethyst` format, that way we wouldn't require an additional format for what is basically an option for an existing one. It could look like this:
```rust
pub struct AmethystFormatOptions {
    pub sprite_naming: SpriteNaming,
}

pub enum SpriteNaming {
    Enumerated, // This is the default option
    Named(Vec<String>),
}
```

Reply:
i knew there must be a better way to share all that boilerplate. i'll look into this for sure.

Reply:
there's an issue with this method. the output of the `Format::encode` function will depend on the `Format::Data` which will be different depending the `SpriteNaming`. so i went ahead and modified the existing method.

Reply:
Alright, sounds good to me.